embassy-net/src/udp.rs

```diff
@@ -122,9 +124,13 @@ impl<'a> UdpSocket<'a> {
     ///
     /// When a datagram is received, this method will return `Poll::Ready` with the
     /// number of bytes received and the remote endpoint.
-    pub fn poll_recv_from(&self, buf: &mut [u8], cx: &mut Context<'_>) -> Poll<Result<(usize, IpEndpoint), RecvError>> {
+    pub fn poll_recv_from(
+        &self,
+        buf: &mut [u8],
+        cx: &mut Context<'_>,
+    ) -> Poll<Result<(usize, UdpMetadata), RecvError>> {
```

Comment:
`recv_from` and `poll_recv_from` should return the same. can you change `recv_from` too?

(the convention is `poll_*` functions are identical to their non-poll counterpart, except manually polled)

Reply:
Done in a fixup; this even makes the total diff smaller. Shall I squash and force-push now or when reviews are done?